test failure with coverage 6.2

# Summary

Looking at updating python-coverage in Fedora Rawhide to 6.2, but getting test failure in python-pytest-cov:
```
============================= test session starts ==============================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
rootdir: /home/orion/fedora/python-pytest-cov/pytest-cov-3.0.0, configfile: setup.cfg, testpaths: tests
plugins: cov-3.0.0, forked-1.3.0, rerunfailures-9.1.1, xdist-2.4.0, arraydiff-0.3, timeout-2.0.1, mpi-0+unknown, remotedata-0.3.2, doctestplus-0.9.0, pyfakefs-4.4.0, mock-3.6.1, openfiles-0.5.0
collected 131 items / 18 deselected / 113 selected

tests/test_pytest_cov.py ............................................... [ 41%]
..........F..ss.................................................s.       [100%]

=================================== FAILURES ===================================
_________________________ test_invalid_coverage_source _________________________

testdir = <Testdir local('/tmp/pytest-of-orion/pytest-0/test_invalid_coverage_source0')>

    def test_invalid_coverage_source(testdir):
        script = testdir.makepyfile(SCRIPT)
        testdir.makeini("""
            [pytest]
            console_output_style=classic
        """)
        result = testdir.runpytest('-v',
                                   '--cov=non_existent_module',
                                   '--cov-report=term-missing',
                                   script)
    
        result.stdout.fnmatch_lines([
            '*10 passed*'
        ])
>       result.stderr.fnmatch_lines([
            'Coverage.py warning: No data was collected.*'
        ])
E       Failed: nomatch: 'Coverage.py warning: No data was collected.*'
E           and: '/usr/lib64/python3.10/site-packages/coverage/inorout.py:520: CoverageWarning: Module non_existent_module was never imported. (module-not-imported)'
E           and: '  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")'
E           and: '/usr/lib64/python3.10/site-packages/coverage/control.py:768: CoverageWarning: No data was collected. (no-data-collected)'
E           and: '  self._warn("No data was collected.", slug="no-data-collected")'
E           and: '/home/orion/rpmbuild/BUILDROOT/python-pytest-cov-3.0.0-2.fc36.x86_64/usr/lib/python3.10/site-packages/pytest_cov/plugin.py:308: CovReportWarning: Failed to generate report: No data to report.'
E           and: ''
E           and: '  warnings.warn(CovReportWarning(message))'
E       remains unmatched: 'Coverage.py warning: No data was collected.*'

/home/orion/fedora/python-pytest-cov/pytest-cov-3.0.0/tests/test_pytest_cov.py:970: Failed
----------------------------- Captured stdout call -----------------------------
running: /usr/bin/python3 -mpytest --basetemp=/tmp/pytest-of-orion/pytest-0/test_invalid_coverage_source0/runpytest-0 -v --cov=non_existent_module --cov-report=term-missing /tmp/pytest-of-orion/pytest-0/test_invalid_coverage_source0/test_invalid_coverage_source.py --basetemp=/tmp/pytest-of-orion/pytest-0/basetemp
     in: /tmp/pytest-of-orion/pytest-0/test_invalid_coverage_source0
============================= test session starts ==============================
platform linux -- Python 3.10.0, pytest-6.2.5, py-1.10.0, pluggy-1.0.0 -- /usr/bin/python3
cachedir: .pytest_cache
rootdir: /tmp/pytest-of-orion/pytest-0/test_invalid_coverage_source0, configfile: tox.ini
plugins: cov-3.0.0, forked-1.3.0, rerunfailures-9.1.1, xdist-2.4.0, arraydiff-0.3, timeout-2.0.1, mpi-0+unknown, remotedata-0.3.2, doctestplus-0.9.0, pyfakefs-4.4.0, mock-3.6.1, openfiles-0.5.0
collecting ... collected 10 items

test_invalid_coverage_source.py::test_foo[0] PASSED
test_invalid_coverage_source.py::test_foo[1] PASSED
test_invalid_coverage_source.py::test_foo[2] PASSED
test_invalid_coverage_source.py::test_foo[3] PASSED
test_invalid_coverage_source.py::test_foo[4] PASSED
test_invalid_coverage_source.py::test_foo[5] PASSED
test_invalid_coverage_source.py::test_foo[6] PASSED
test_invalid_coverage_source.py::test_foo[7] PASSED
test_invalid_coverage_source.py::test_foo[8] PASSED
test_invalid_coverage_source.py::test_foo[9] PASSEDWARNING: Failed to generate report: No data to report.



---------- coverage: platform linux, python 3.10.0-final-0 -----------


============================== 10 passed in 0.06s ==============================
----------------------------- Captured stderr call -----------------------------
/usr/lib64/python3.10/site-packages/coverage/inorout.py:520: CoverageWarning: Module non_existent_module was never imported. (module-not-imported)
  self.warn(f"Module {pkg} was never imported.", slug="module-not-imported")
/usr/lib64/python3.10/site-packages/coverage/control.py:768: CoverageWarning: No data was collected. (no-data-collected)
  self._warn("No data was collected.", slug="no-data-collected")
/home/orion/rpmbuild/BUILDROOT/python-pytest-cov-3.0.0-2.fc36.x86_64/usr/lib/python3.10/site-packages/pytest_cov/plugin.py:308: CovReportWarning: Failed to generate report: No data to report.

  warnings.warn(CovReportWarning(message))
=============================== warnings summary ===============================
../../../../../usr/lib/python3.10/site-packages/_pytest/config/__init__.py:1183
  /usr/lib/python3.10/site-packages/_pytest/config/__init__.py:1183: PytestDeprecationWarning: The --strict option is deprecated, use --strict-markers instead.
    self.issue_config_time_warning(

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
SKIPPED [1] tests/test_pytest_cov.py:1047: deadlocks on Python 3.8+, see: https://bugs.python.org/issue38227
SKIPPED [1] tests/test_pytest_cov.py:1089: deadlocks on Python 3.8, see: https://bugs.python.org/issue38227
SKIPPED [1] tests/test_pytest_cov.py:2119: condition: coverage.version_info >= (5, 0)
FAILED tests/test_pytest_cov.py::test_invalid_coverage_source - Failed: nomat...
= 1 failed, 109 passed, 3 skipped, 18 deselected, 1 warning in 278.85s (0:04:38) =
```
It does appear to work with coverage 5.6 which is currently in Fedora Rawhide.

--- a/tests/test_pytest_cov.py
+++ b/tests/test_pytest_cov.py
@@ -981,7 +981,7 @@ def test_invalid_coverage_source(testdir):
         '*10 passed*'
     ])
     result.stderr.fnmatch_lines([
-        'Coverage.py warning: No data was collected.*'
+        '*No data was collected.*'
     ])
     result.stdout.fnmatch_lines([
         '*Failed to generate report: No data to report.',

